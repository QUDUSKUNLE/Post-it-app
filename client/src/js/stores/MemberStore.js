import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher.js';

import {
  GENERAL, ADD_MEMBER, GET_MEMBERS_OF_A_GROUP } from
  '../constants/ActionConstants.js';

/**
 * Holds the storage, listen to actions and update the stores
 * @class MemberStore
 */
class MemberStore extends EventEmitter {
  /**
   * sets the members, general to an empty []
   * @constructor
   */
  constructor() {
    super();
    this.members = [];
    this.group = '';
    this.general = [];
    this.allGroupMembers = this.allGroupMembers.bind(this);
    this.allGeneralUsers = this.allGeneralUsers.bind(this);
    this.handleActions = this.handleActions.bind(this);
    this.setGroupMembers = this.setGroupMembers.bind(this);
  }

  /**
   * @method allGeneralUsers
   * @return {object} generalmembers - The general stored in the constructor
   */
  allGeneralUsers() {
    return this.general;
  }

  /**
   * @method allGroupMembers
   * @return {object} [members & group] - members and group stored in the
    constructor
   */
  allGroupMembers() {
    return [this.members, this.group];
  }

  /**
   * @method setGroupMembers
   * @param {Object} group -
   * @return {object} update mmebers - The groups stored in the constructor
   */
  setGroupMembers(group) {
    this.members = group;
  }

  /**
   * Receives actions and update the stores accordingly
   * @method handleActions
   * @param {object} action - Action type and data
   * @return {null} -
   */
  handleActions(action) {
    switch (action.type) {
      case GENERAL:
        this.general = action.general;
        this.emit('GENERAL');
        break;

      case GET_MEMBERS_OF_A_GROUP:
        this.members = action.members;
        this.group = action.group;
        this.emit('GET_MEMBERS_OF_A_GROUP');
        break;

      case ADD_MEMBER:
        this.members.push(action.members);
        this.emit('ADD_MEMBER');
        break;

      default:
    }
  }
}

// create a new instance of MemberStore
const memberStore = new MemberStore();

// register an AppDispatcher and bind it to handleActions method
AppDispatcher.register(memberStore.handleActions.bind(memberStore));

// export an instance of MemberStore
export default memberStore;
