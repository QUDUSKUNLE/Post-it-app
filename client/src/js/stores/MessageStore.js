import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher.js';

import {
  GENERAL_MESSAGES, ADD_MESSAGE,
  GET_GROUP_MESSAGES, SEND_MESSAGE } from
  '../constants/ActionConstants.js';


class MessageStore extends EventEmitter {
  constructor(props) {
    super(props);
    this.message = [];
    this.generalMessages = [];
    this.groupMessages = [];
    this.allGroupMessages = this.allGroupMessages.bind(this);
    this.allGeneralMessages = this.allGeneralMessages.bind(this);
    this.handleActions = this.handleActions.bind(this);
  }

  allGeneralMessages() {
    return this.general;
  }

  allGroupMessages() {
    return this.members;
  }

  sendMessage() {
    return this.message;
  }

  handleActions(action) {
    switch (action.type) {
      case GENERAL_MESSAGES:
        this.general = action.general;
        this.emit('GENERAL_MESSAGES');
        break;

      case GET_GROUP_MESSAGES:
        this.members = action.members;
        this.emit('GET_GROUP_MESSAGES');
        break;

      case ADD_MESSAGE:
        this.members.push(action.members);
        this.emit('ADD_MESSAGE');
        break;

      case SEND_MESSAGE:
        this.message.push(action.message);
        this.emit('SEND_MESSAGE');
        break;

      default:
    }
  }
}

const messageStore = new MessageStore();

AppDispatcher.register(messageStore.handleActions.bind(messageStore));

export default messageStore;
