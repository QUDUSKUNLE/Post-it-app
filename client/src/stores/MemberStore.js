import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher';
import { getGroupMembers } from '../helper/convertObject';
import {
  ADD_MEMBER,
  GET_MEMBERS_OF_GROUP, SEARCH_USER } from '../constants/ActionConstants';

/**
 * Holds the storage, listen to actions and update the stores
 * @class MemberStore
 */
class MemberStore extends EventEmitter {
  /**
   * sets the members, general to an empty []
   * @constructor
   */
  constructor() {
    super();
    this.groupIndex = [];
    this.members = [];
    this.group = '';
    this.groupId = '';
    this.searchUser = {};
    this.addMemberResponse = '';
    this.getSearchUser = this.getSearchUser.bind(this);
    this.allGroupMembers = this.allGroupMembers.bind(this);
    this.addMember = this.addMember.bind(this);
    this.handleActions = this.handleActions.bind(this);
  }

  /**
   * @method getSearchUser
   * @return {object} searchUser - searchUser stored in the constructor method
   */
  getSearchUser() {
    return [this.searchUser];
  }

  /**
   * @method allGroupMembers
   * @return {object} [members & group] - members and group stored in the
    constructor
   */
  allGroupMembers() {
    return [this.members, this.group, this.groupId];
  }

  /**
   * @method addMember
   * @return {string} addMemberResponse - addMemberResponse in the store
   */
  addMember() {
    return this.addMemberResponse;
  }

  /**
   * Receives actions and update the stores accordingly
   * @method handleActions
   * @param {object} action - Action type and data
   * @return {null} -
   */
  handleActions(action) {
    switch (action.type) {
      case GET_MEMBERS_OF_GROUP:
        this.groupIndex = action.members;
        this.members = getGroupMembers(this.groupIndex);
        this.groupId = (action.members)[1];
        this.group = (action.members)[2];
        this.emit('GET_MEMBERS_OF_GROUP');
        break;

      case SEARCH_USER:
        this.searchUser = action.search;
        this.emit('SEARCH_USER');
        break;

      case ADD_MEMBER:
        this.addMemberResponse = action.member;
        this.emit('ADD_MEMBER');
        break;
      default:
    }
  }
}

// create a new instance of MemberStore
const memberStore = new MemberStore();

// register an AppDispatcher and bind it to handleActions method
AppDispatcher.register(memberStore.handleActions.bind(memberStore));

// export an instance of MemberStore
export default memberStore;
