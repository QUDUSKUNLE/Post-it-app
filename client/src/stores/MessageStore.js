import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher';

import {
  GET_GROUP_MESSAGE, SEND_GROUP_MESSAGE } from
  '../constants/ActionConstants';


/**
 * @class MessageStore
 * @extends {EventEmitter}
 */
class MessageStore extends EventEmitter {
  /**
   * Creates an instance of MessageStore.
   * @memberof MessageStore
   */
  constructor() {
    super();
    this.groupMessage = [];
    this.allGroupMessage = this.allGroupMessage.bind(this);
    this.handleActions = this.handleActions.bind(this);
  }

  /**
   * @returns {object} - groupMessage
   * @memberof MessageStore
   */
  allGroupMessage() {
    return this.groupMessage;
  }

  /**
   * @param {any} action - payload
   * @memberof MessageStore
   * @return {*} action
   */
  handleActions(action) {
    switch (action.type) {
      case GET_GROUP_MESSAGE:
        this.groupMessage = action.message;
        this.emit('GET_GROUP_MESSAGE');
        break;

      case SEND_GROUP_MESSAGE:
        this.groupMessage.push(action.message);
        this.emit('SEND_GROUP_MESSAGE');
        break;

      default:
    }
  }
}

// create a new instance of MessageStore
const messageStore = new MessageStore();

// register an AppDispatcher and bind it to handleActions method
AppDispatcher.register(messageStore.handleActions.bind(messageStore));

// export an instance of MessageStore
export default messageStore;
