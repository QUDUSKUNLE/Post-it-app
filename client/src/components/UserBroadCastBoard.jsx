import React from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import $ from 'jquery';
import { Link, Redirect } from 'react-router-dom';
import UserGroups from './UserGroups';
import UserChatBox from './UserChatBox';
import Footer from './Footer';
import NoGroupSelected from './NoGroupSelected';
import { getUserGroups } from '../actions/groupAction';
import { getGroupMessage } from '../actions/messageActions';
import { signoutAction } from '../actions/signOutActions';
import { getGroupMember } from '../actions/memberActions';
import MemberStore from '../stores/MemberStore';
import GroupStore from '../stores/GroupStore';
import MessageStore from '../stores/MessageStore';

/**
 * @description - renders BroadCastBoard Component
 * @class BroadCastBoard
 * @extends {React.Component}
 */
export default class UserBroadCastBoard extends React.Component {
  /**
   * Create a constructor
   * @constructor
   * @param {object} props -
   */
  constructor(props) {
    super(props);
    this.state = {
      loggedIn: JSON.parse(localStorage.getItem('userIn')),
      userId: JSON.parse(localStorage.getItem('Id')),
      defaultGroup: '',
      groups: [],
      groupId: '',
      userName: JSON.parse(localStorage.getItem('userName')),
      groupMessage: [],
      groupMember: [],
      groupSelected: false
    };

    /**
     * @description This binding is necessary to make `this` work
     * in the callback
     */
    this.handleOnChangeEvent = this.handleOnChangeEvent.bind(this);
    this.handleSignOutAction = this.handleSignOutAction.bind(this);
    this.handleSendGroupMessage = this.handleSendGroupMessage.bind(this);
    this.handleGetGroupMessage = this.handleGetGroupMessage.bind(this);
    this.handleGetUserGroups = this.handleGetUserGroups.bind(this);
    this.handleGetGroupMember = this.handleGetGroupMember.bind(this);
  }

  /**
   * Call action on initial page load
   * @method componentWillMount
   * @return {*} void
   */
  componentWillMount() {
    getUserGroups(this.state.userId);
  }

  /**
   * Attach an event listener to favorite store
   * @method componentDidMount
   * @return {*} -
   */
  componentDidMount() {
    GroupStore.on('GET_USER_GROUPS', this.handleGetUserGroups);
    MemberStore.on('GET_MEMBERS_OF_GROUP', this.handleGetGroupMember);
    MessageStore.on('SEND_GROUP_MESSAGE', this.handleSendGroupMessage);
    MessageStore.on('GET_GROUP_MESSAGE', this.handleGetGroupMessage);
  }

  /**
   * @method componentWillUnount
   * @return {*} void
   */
  componentWillUnmount() {
    GroupStore.removeListener('GET_USER_GROUPS',
      this.handleGetUserGroups);
    MemberStore.removeListener('GET_MEMBERS_OF_GROUP',
      this.handleGetGroupMember);
    MessageStore.removeListener('SEND_GROUP_MESSAGE',
      this.handleSendGroupMessage);
    MessageStore.removeListener('GET_GROUP_MESSAGE',
      this.handleGetGroupMessage);
  }

  /**
   * @method newGroupMessage
   * @return {*} void
   */
  handleSendGroupMessage() {
    this.setState({
      groupMessage: MessageStore.allGroupMessage()
    });
  }

  handleGetGroupMember() {
    this.setState({ groupMember: (MemberStore.allGroupMembers())[0] });
  }

  /**
   * @method getGroupMessage
   * @return {*} void
   */
  handleGetGroupMessage() {
    this.setState({ groupMessage: MessageStore.allGroupMessage() });
  }

  /**
   * @method userGroups
   * @return {*} void
   */
  handleGetUserGroups() {
    this.setState({ groups: GroupStore.allGroups() });
  }
  /**
   * onChange event.
   * @param {object} event The first number.
   * @returns {*} bind input values to name.
   */
  handleOnChangeEvent(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSignOutAction() {
    signoutAction().then((response) => {
      toastr.success(response.data.message);
      this.props.history.push('/');
      localStorage.clear();
      location.reload();
    }).catch((error) => toastr.error(error.response.data));
  }

  /**
   * @description - render method, React lifecycle method
   * @returns {*} BroadCastBoard component
   */
  render() {
    if (!this.state.loggedIn) {
      return (
        <Redirect to="/signin" />
      );
    }
    const groupList = this.state.groups.map((group) =>
      <li
        className="text-center"
        key={Object.values(group)}
        value={Object.keys(group)}
        name={Object.keys(group)}
        onClick={() => {
          getGroupMember(Object.values(group));
          getGroupMessage(Object.values(group));
          this.setState({
            defaultGroup: (Object.keys(group))[0],
            groupId: (Object.values(group))[0],
            groupSelected: true
          });
        }}><Link to="#"> {Object.keys(group)}</Link>
      </li>);
    const isGroupSelected = () => {
      let selectedGroup;
      if (this.state.groupSelected) {
        selectedGroup =
          (<UserChatBox
            defaultGroup={this.state.defaultGroup}
            groupId={this.state.groupId}
            allGeneralMessage={this.state.groupMessage} />);
      } else {
        selectedGroup = <NoGroupSelected />;
      }
      return selectedGroup;
    };
    $(document).ready(() => {
      $('[data-toggle=offcanvas]').click(() => {
        $('.row-offcanvas').toggleClass('active');
      });
    });
    return (
      <div>
        <div className="navbar navbar-inverse navabar-fixed-top">
          <div className="container-fluid">
            <div className="navbar-header">
              <button type="button" className="navbar-toggle"
                data-toggle="collapse" data-target=".navbar-collapse">
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
              </button>
              <Link className="navbar-brand" to="#">
                PostIt<small>App</small>
              </Link>
            </div>
            <div className="collapse navbar-collapse">
              <ul className="nav navbar-nav navbar-right">
                <li className="active">
                  <Link to="/broadcastboard">MessageBoard</Link>
                </li>
                <li>
                  <Link to="/group">Create Group</Link>
                </li>
                <li onClick={this.handleSignOutAction}>
                  <Link to="#">Sign out</Link>
                </li>
              </ul>
            </div>
          </div>
        </div>
        {/* <div className="row">
          <div className="col-md-12">
            <p className="pull-left">{`Hi, ${this.state.userName}.`}</p>
          </div>
        </div> */}
        <div className="row-offcanvas row-offcanvas-left">
          <UserGroups
            grouplist={groupList}
            member={this.state.groupMember}/>
          {isGroupSelected()}
        </div>
        <Footer/>
      </div>
    );
  }
}

// props validation
UserBroadCastBoard.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  })
};
