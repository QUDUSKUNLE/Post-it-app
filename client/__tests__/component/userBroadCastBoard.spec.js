import React from 'react';
import PropTypes from 'prop-types';
import { mount, shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import expect from 'expect';
import sinon from 'sinon';
import 'babel-polyfill';
import MessageStore from '../../src/stores/MessageStore';
import MemberStore from '../../src/stores/MemberStore';
import localStorageMock from '../../src/__mock__/localStorage';
import UserBroadCastBoard from '../../src/components/UserBroadCastBoard';

window.localStorage = localStorageMock;
describe('<UserBroadCastBoard/>', () => {
  let wrapper;
  let component;
  const mockOnMessage = sinon.stub(MessageStore,
    'on').callsFake((message, cb) => cb());
  const mockUnMountMessage = sinon.stub(MessageStore,
    'removeListener').callsFake((message, cb) => cb());
  const mockMessageResponse = sinon.stub(MessageStore,
    'allGroupMessage').returns([{ name: 'Gold', message: 'Shola is here' }]);

  const mockOnMembers = sinon.stub(MemberStore,
    'on').callsFake((member, cb) => cb());
  const mockUnMountMembers = sinon.stub(MemberStore,
    'removeListener').callsFake((member, cb) => cb());
  const mockMemberResponse = sinon.stub(MemberStore,
    'allGroupMembers').returns([[{ a: 1 }, { b: 2 }, { c: 3 }]]);

  beforeEach(() => {
    window.localStorage.setItem('userIn', JSON.stringify(true));
    window.localStorage.setItem('Id', JSON.stringify('AZCVGFRTUINSMUY15156'));
    window.localStorage.setItem('userName', JSON.stringify('Kunle'));
    const props = {
      loggedIn: JSON.parse(localStorage.getItem('userIn')),
      userId: JSON.parse(localStorage.getItem('Id')),
      defaultGroup: '',
      groups: [],
      groupId: '',
      userName: JSON.parse(localStorage.getItem('userName')),
      groupMessage: [],
      groupMember: [],
      groupSelected: false
    };
    component = shallow(<UserBroadCastBoard {...props}/>);
    wrapper = mount(<UserBroadCastBoard {...props}/>,
      {
        childContextTypes: { router: PropTypes.object },
        context: {
          router:
          {
            history: {
              push: () => null,
              replace: () => null,
              createHref: () => null,
              path: '/broadcastboard',
              component: '[function UserBroadCastBoard]',
              location: {
                pathname: '/broadcastboard',
                search: '',
                hash: '',
                key: '6l9jpq'
              },
              computedMatch: {
                path: '/broadcastboard',
                url: '/broadcastboard',
                isExact: true,
                params: {}
              }
            }
          }
        }
      }
    );
  });
  it('component expected to be defined', () => {
    expect(UserBroadCastBoard).toBeDefined();
    expect(wrapper.state()).toBeDefined();
    expect(wrapper.state().userName).toEqual('Kunle');
    expect(wrapper.state().userId).toEqual('AZCVGFRTUINSMUY15156');
    expect(wrapper.state().groupSelected).toEqual(false);
    expect(wrapper.state().defaultGroup).toEqual('');
    expect(wrapper.state().groups).toEqual([]);
    expect(wrapper.state().groupId).toEqual('');
  });

  it('component should contain defined methods', () => {
    expect(wrapper.nodes[0].handleSignOutAction).toBeDefined();
    expect(wrapper.nodes[0].handleSendGroupMessage).toBeDefined();
    expect(wrapper.nodes[0].handleGetGroupMessage).toBeDefined();
    expect(wrapper.nodes[0].handleGetUserGroups).toBeDefined();
    expect(wrapper.nodes[0].handleGetGroupMember).toBeDefined();
  });
  it('should render correctly without crashing', () => {
    expect(component).toMatchSnapshot();
    expect(component.find('div')).toHaveLength(6);
    expect(component.find('span')).toHaveLength(3);
    expect(component.find('form')).toHaveLength(0);
    expect(component.find('button')).toHaveLength(1);
    expect(component.find('Link')).toHaveLength(4);
    expect(component.find('Footer')).toHaveLength(1);
  });
  it('should find it Links to other paths', () => {
    expect(wrapper.find(Link).at(1).prop('to')).toEqual('/broadcastboard');
    expect(wrapper.find(Link).at(2).prop('to')).toEqual('/group');
  });
  it('component should call componentDidMount component lifecycle', () => {
    const spy = sinon.spy(UserBroadCastBoard.prototype, 'componentDidMount');
    wrapper.instance().componentDidMount();
    expect(spy.calledOnce).toBeTruthy();
  });
  it('component should call componentWillUnmount component lifecycle', () => {
    const spy = sinon.spy(UserBroadCastBoard.prototype, 'componentWillUnmount');
    wrapper.instance().componentWillUnmount();
    expect(spy.calledOnce).toBeTruthy();
  });
  it('expects componentDidMount to mount when event is fired', () => {
    expect(mockOnMessage.displayName).toEqual('on');
    expect(wrapper.state().loggedIn).toEqual(true);
    expect(wrapper.state().userId).toEqual('AZCVGFRTUINSMUY15156');
    expect(wrapper.state().groupMessage).toEqual(mockMessageResponse());
  });
  it('expects componentDidMount to mount', () => {
    expect(mockOnMembers.displayName).toEqual('on');
    expect(wrapper.state().groupMember).toEqual(mockMemberResponse()[0]);
  });
  it('expects componentWillUnmount to be unmounted', () => {
    wrapper.unmount();
    expect(mockUnMountMessage.callCount).toBe(4);
    expect(mockUnMountMessage.displayName).toEqual('removeListener');
  });
  it('expects componentWillUnmount displayName to be removeListener', () => {
    wrapper.unmount();
    expect(mockUnMountMembers.displayName).toEqual('removeListener');
  });
});
