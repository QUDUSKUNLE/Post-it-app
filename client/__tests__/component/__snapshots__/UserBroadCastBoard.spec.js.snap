// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBroadCastBoard Component should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="navbar navbar-inverse navabar-fixed-top"
    >
      <div
        className="container-fluid"
      >
        <div
          className="navbar-header"
        >
          <button
            className="navbar-toggle"
            data-target=".navbar-collapse"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
          <Link
            className="navbar-brand"
            replace={false}
            to="#"
          >
            PostIt
            <small>
              App
            </small>
          </Link>
        </div>
        <div
          className="collapse navbar-collapse"
        >
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li
              className="active"
            >
              <Link
                replace={false}
                to="/broadcastboard"
              >
                MessageBoard
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/group"
              >
                Create Group
              </Link>
            </li>
            <li
              onClick={[Function]}
            >
              <Link
                replace={false}
                to="#"
              >
                Sign out
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className="row"
      >
        <div
          className="col-sm-12 col-md-2"
        >
          <UserGroups
            grouplist={Array []}
            member={Array []}
          />
        </div>
        <div
          className="col-sm-12 col-md-10"
        >
          <NoGroupSelected />
        </div>
      </div>
      <div
        className="row"
      >
        <Footer />
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        className="navbar navbar-inverse navabar-fixed-top"
      >
        <div
          className="container-fluid"
        >
          <div
            className="navbar-header"
          >
            <button
              className="navbar-toggle"
              data-target=".navbar-collapse"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
            </button>
            <Link
              className="navbar-brand"
              replace={false}
              to="#"
            >
              PostIt
              <small>
                App
              </small>
            </Link>
          </div>
          <div
            className="collapse navbar-collapse"
          >
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li
                className="active"
              >
                <Link
                  replace={false}
                  to="/broadcastboard"
                >
                  MessageBoard
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/group"
                >
                  Create Group
                </Link>
              </li>
              <li
                onClick={[Function]}
              >
                <Link
                  replace={false}
                  to="#"
                >
                  Sign out
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className="row"
        >
          <div
            className="col-sm-12 col-md-2"
          >
            <UserGroups
              grouplist={Array []}
              member={Array []}
            />
          </div>
          <div
            className="col-sm-12 col-md-10"
          >
            <NoGroupSelected />
          </div>
        </div>
        <div
          className="row"
        >
          <Footer />
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserBroadCastBoard
        defaultGroup=""
        groupId=""
        groupMember={Array []}
        groupMessage={Array []}
        groupSelected={false}
        groups={Array []}
        loggedIn={true}
        userId="AZCVGFRTUINSMUY15156"
        userName="Kunle"
      />,
      "_debugID": 99,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserBroadCastBoard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleGetGroupMember": [Function],
        "handleGetGroupMessage": [Function],
        "handleGetUserGroups": [Function],
        "handleOnChangeEvent": [Function],
        "handleSendGroupMessage": [Function],
        "handleSignOutAction": [Function],
        "props": Object {
          "defaultGroup": "",
          "groupId": "",
          "groupMember": Array [],
          "groupMessage": Array [],
          "groupSelected": false,
          "groups": Array [],
          "loggedIn": true,
          "userId": "AZCVGFRTUINSMUY15156",
          "userName": "Kunle",
        },
        "refs": Object {},
        "state": Object {
          "defaultGroup": "",
          "groupId": "",
          "groupMember": Array [],
          "groupMessage": Array [],
          "groupSelected": false,
          "groups": Array [],
          "loggedIn": true,
          "userId": "AZCVGFRTUINSMUY15156",
          "userName": "Kunle",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            className="navbar navbar-inverse navabar-fixed-top"
          >
            <div
              className="container-fluid"
            >
              <div
                className="navbar-header"
              >
                <button
                  className="navbar-toggle"
                  data-target=".navbar-collapse"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                </button>
                <Link
                  className="navbar-brand"
                  replace={false}
                  to="#"
                >
                  PostIt
                  <small>
                    App
                  </small>
                </Link>
              </div>
              <div
                className="collapse navbar-collapse"
              >
                <ul
                  className="nav navbar-nav navbar-right"
                >
                  <li
                    className="active"
                  >
                    <Link
                      replace={false}
                      to="/broadcastboard"
                    >
                      MessageBoard
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/group"
                    >
                      Create Group
                    </Link>
                  </li>
                  <li
                    onClick={[Function]}
                  >
                    <Link
                      replace={false}
                      to="#"
                    >
                      Sign out
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            className=""
          >
            <div
              className="row"
            >
              <div
                className="col-sm-12 col-md-2"
              >
                <UserGroups
                  grouplist={Array []}
                  member={Array []}
                />
              </div>
              <div
                className="col-sm-12 col-md-10"
              >
                <NoGroupSelected />
              </div>
            </div>
            <div
              className="row"
            >
              <Footer />
            </div>
          </div>
        </div>,
        "_debugID": 100,
        "_renderedOutput": <div>
          <div
            className="navbar navbar-inverse navabar-fixed-top"
          >
            <div
              className="container-fluid"
            >
              <div
                className="navbar-header"
              >
                <button
                  className="navbar-toggle"
                  data-target=".navbar-collapse"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                </button>
                <Link
                  className="navbar-brand"
                  replace={false}
                  to="#"
                >
                  PostIt
                  <small>
                    App
                  </small>
                </Link>
              </div>
              <div
                className="collapse navbar-collapse"
              >
                <ul
                  className="nav navbar-nav navbar-right"
                >
                  <li
                    className="active"
                  >
                    <Link
                      replace={false}
                      to="/broadcastboard"
                    >
                      MessageBoard
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/group"
                    >
                      Create Group
                    </Link>
                  </li>
                  <li
                    onClick={[Function]}
                  >
                    <Link
                      replace={false}
                      to="#"
                    >
                      Sign out
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            className=""
          >
            <div
              className="row"
            >
              <div
                className="col-sm-12 col-md-2"
              >
                <UserGroups
                  grouplist={Array []}
                  member={Array []}
                />
              </div>
              <div
                className="col-sm-12 col-md-10"
              >
                <NoGroupSelected />
              </div>
            </div>
            <div
              className="row"
            >
              <Footer />
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserBroadCastBoard
    defaultGroup=""
    groupId=""
    groupMember={Array []}
    groupMessage={Array []}
    groupSelected={false}
    groups={Array []}
    loggedIn={true}
    userId="AZCVGFRTUINSMUY15156"
    userName="Kunle"
  />,
}
`;
